modules/programs/hexchat.nix

```diff
@@ -0,0 +1,256 @@
+{ config, pkgs, lib, ... }:
+with builtins;
+with lib;
+let cfg = config.programs.hexchat;
+in {
+  meta.maintainers = [ maintainers.superherointj ];
+
+  options.programs.hexchat = {
+    channels = with lib.types;
+      let
+        modChannelOption = submodule {
+          options = {
+            options = let
+              channelOptions = submodule {
+                options = {
+                  autoconnect = mkOption {
+                    type = nullOr bool;
+                    description = "Autoconnect to network";
+                    default = false;
+                  };
+                  connect_to_selected_server_only = mkOption {
+                    type = nullOr bool;
+                    description = "Connect to selected server only";
+                    default = true;
+                  };
+                  bypass_proxy = mkOption {
+                    type = nullOr bool;
+                    description = "Bypass proxy";
+                    default = true;
+                  };
+                  use_ssl_for_all_servers = mkOption {
+                    type = nullOr bool;
+                    description = "Use SSL for all servers";
+                    default = false;
+                  };
+                  accept_invalid_ssl_certificates = mkOption {
+                    type = nullOr bool;
+                    description = "Accept invalid SSL certificates";
+                    default = false;
+                  };
+                  use_global_user_information = mkOption {
+                    type = nullOr bool;
+                    description = "Use global user information";
+                    default = false;
+                  };
+                };
+              };
+            in mkOption {
+              type = nullOr (channelOptions);
+              default = null;
+              description = "";
+              example = "";
+            };
+            nickname = mkOption {
+              type = str;
+              description = "Primary nickname";
+              defaultText = "nickname1";
```

Comment:
I think you meant `default` here instead? Anyway, I wouldn't set any default for this case.